include/boost/geometry/formulas/distance_point_segment.hpp

```diff
@@ -0,0 +1,335 @@
+// Boost.Geometry
+
+// Copyright (c) 2016-2017 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_CROSS_TRACK_GEO_HPP
+#define BOOST_GEOMETRY_FORMULAS_CROSS_TRACK_GEO_HPP
+
+#include <boost/geometry/formulas/result_direct.hpp>
+
+#ifndef BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS
+#define BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS 100
+#endif
+
+/*!
+\brief Algorithm to compute the distance between a segment and a point using
+       direct and inverse geodesic problems as subroutines. The algorithm
+       approximates the distance by an iterative Newton method.
+\see C.F.F.Karney - Geodesics on an ellipsoid of revolution,
+      https://arxiv.org/abs/1102.1215
+*/
+
+namespace boost { namespace geometry { namespace formula
+{
+
+template
+<
+        typename CT,
+        typename Units,
+        typename Inverse_type_azimuth,
+        typename Inverse_type_distance,
+        typename Direct_type,
```

Comment:
I suggest to take direct and inverse formulas as class template instead of a type here:

    template <typename, bool, bool, bool, bool ,bool> class Inverse,
    template <typename, bool, bool, bool, bool> class Direct,

And then internally define types passing appropriate parameters:

    typedef Inverse<CT, true, false, false, false, false> inverse_distance_type;
    typedef Inverse<CT, false, true, false, false, false> inverse_azimuth_type;

This way this formula would create inverse formula types able to calculate whatever is needed, so it wouldn't rely on a caller. Currently the caller has to know which results of inverse formula will be used internally and pass the correct type.